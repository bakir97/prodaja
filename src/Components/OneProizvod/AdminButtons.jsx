import React from "react";
import { Grid, Button, Typography } from "@material-ui/core";
import Edit from "@material-ui/icons/Edit";
import Delete from "@material-ui/icons/Delete";
const styles = {
  button: {
    fontFamily: "Kalam",
    paddingBottom: 0,
    marginRight: "1rem",
    WebkitBackgroundClip: "text",
    color: "transparent"
  }
};
const AdminButtons = ({ history, match, handleDelete }) => {
  return (
    <Grid
      container
      justify="center"
      alignItems="center"
      style={{ marginTop: "1rem" }}
    >
      <Button onClick={() => history.push(`/Edit/${match.params.id}`)}>
        <Typography
          variant="h5"
          style={{
            background: "linear-gradient(to right, #fdc830, #f37335)",
            ...styles.button
          }}
        >
          Edit
        </Typography>
        <Edit
          style={{
            color: "#f37335",
            marginBottom: 5
          }}
        />
      </Button>
      <Button onClick={() => handleDelete()}>
        <Typography
          variant="h5"
          style={{
            background: "linear-gradient(to right, #333333, #dd1818)",
            ...styles.button
          }}
        >
          Delete
        </Typography>
        <Delete
          style={{
            color: "#dd1818",
            marginBottom: 5
          }}
        />
      </Button>
    </Grid>
  );
};

export default AdminButtons;
