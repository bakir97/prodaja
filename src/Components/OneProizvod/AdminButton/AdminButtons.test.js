import AdminButtons from "./index";
describe("AdminButtons", () => {
  const params = {
    history: {
      push: jest.fn()
    },
    match: { params: { id: 1 } },
    handleDelete: jest.fn()
  };
  const wrapper = shallow(<AdminButtons {...params} />);
  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("should push to edit", () => {
    wrapper.find("#edit").simulate("click");
    expect(params.history.push).toBeCalledWith("/Edit/1");
  });
  it("should call delete", () => {
    wrapper.find("#delete").simulate("click");
    expect(params.handleDelete).toBeCalled();
  });
});
