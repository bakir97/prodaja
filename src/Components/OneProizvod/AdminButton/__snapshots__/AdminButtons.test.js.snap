// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminButtons should render 1`] = `
<WithStyles(Grid)
  alignItems="center"
  container={true}
  justify="center"
  style={
    Object {
      "marginTop": "1rem",
    }
  }
>
  <WithStyles(Button)
    id="edit"
    onClick={[Function]}
  >
    <WithStyles(Typography)
      style={
        Object {
          "WebkitBackgroundClip": "text",
          "background": "linear-gradient(to right, #fdc830, #f37335)",
          "color": "transparent",
          "fontFamily": "Kalam",
          "marginRight": "1rem",
          "paddingBottom": 0,
        }
      }
      variant="h5"
    >
      Edit
    </WithStyles(Typography)>
    <pure(EditIcon)
      style={
        Object {
          "color": "#f37335",
          "marginBottom": 5,
        }
      }
    />
  </WithStyles(Button)>
  <WithStyles(Button)
    id="delete"
    onClick={[Function]}
  >
    <WithStyles(Typography)
      style={
        Object {
          "WebkitBackgroundClip": "text",
          "background": "linear-gradient(to right, #333333, #dd1818)",
          "color": "transparent",
          "fontFamily": "Kalam",
          "marginRight": "1rem",
          "paddingBottom": 0,
        }
      }
      variant="h5"
    >
      Delete
    </WithStyles(Typography)>
    <pure(DeleteIcon)
      style={
        Object {
          "color": "#dd1818",
          "marginBottom": 5,
        }
      }
    />
  </WithStyles(Button)>
</WithStyles(Grid)>
`;
